platform/ios/MGLMapView.mm

```diff
@@ -172,9 +172,18 @@ - (void)setStyleJSON:(NSString *)styleJSON
     }
 }
 
-- (void)setStyleURL:(NSString *)filePathURL
+- (void)setStyleURL:(NSString *)styleURLString
 {
-    mbglMap->setStyleURL(std::string("asset://") + [filePathURL UTF8String]);
+    std::string styleURL([styleURLString UTF8String]);
+
+    if ( ! [[NSURL URLWithString:styleURLString] scheme])
```

Comment:
It may have no meaning outside of our codebase, but we use to to make our styles compatible across renderers.